### ISSUE TITLE
Issue with Input type and weight type should be the same

### ISSUE THREAD

**Issue by Ca-ressemble-a-du-fake** [Edresson/YourTTS#28]
Hi,

I am trying to train YourTTS on my own dataset. So I followed your helpful [guide](https://github.com/Edresson/YourTTS#reproducibility) with the latest stable version of Coqui TTS (0.8.0).

After computing the embeddings (on GPU) without issue, I run into this `RuntimeError: Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same`.

I have already trained a VITS model with this dataset so everything is already set up. I understood that input Tensor resides on GPU whereas weight Tensor resides on CPU but how can I solve this ? Should I downgrade to CoquiTTS 0.6.2 ?

Here is the full traceback : 
```
File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 1533, in fit
    self._fit()
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 1517, in _fit
    self.train_epoch()
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 1282, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 1135, in train_step
    outputs, loss_dict_new, step_time = self._optimize(
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 996, in _optimize
    outputs, loss_dict = self._model_train_step(batch, model, criterion, optimizer_idx=optimizer_idx)
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/trainer/trainer.py", line 954, in _model_train_step
    return model.train_step(*input_args)
  File "/home/caraduf/YourTTS/TTS/TTS/tts/models/vits.py", line 1250, in train_step
    outputs = self.forward(
  File "/home/caraduf/YourTTS/TTS/TTS/tts/models/vits.py", line 1049, in forward
    pred_embs = self.speaker_manager.encoder.forward(wavs_batch, l2_norm=True)
  File "/home/caraduf/YourTTS/TTS/TTS/encoder/models/resnet.py", line 169, in forward
    x = self.torch_spec(x)
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/torch/nn/modules/container.py", line 139, in forward
    input = module(input)
  File "/home/caraduf/YourTTS/yourtts_env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/caraduf/YourTTS/TTS/TTS/encoder/models/base_encoder.py", line 22, in forward
    return torch.nn.functional.conv1d(x, self.filter).squeeze(1)
RuntimeError: Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same
```
Thanks for helping me out!

**Comment by pbst7** [comment]
If anybody is struggling on this, check this out https://github.com/coqui-ai/TTS/pull/2054

**Comment by MarkuzK** [comment]
Hi @Ca-ressemble-a-du-fake,

Did you resolve the matter? I just ran into almost the identical issue with the aforementioned patch applied. Main difference is that the two tensors are both on cuda in my case:

Traceback (most recent call last):
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 1538, in fit
    self._fit()
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 1522, in _fit
    self.train_epoch()
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 1287, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 1140, in train_step
    outputs, loss_dict_new, step_time = self._optimize(
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 1001, in _optimize
    outputs, loss_dict = self._model_train_step(batch, model, criterion, optimizer_idx=optimizer_idx)
  File "/home/ubuntu/venv/lib/python3.8/site-packages/trainer/trainer.py", line 959, in _model_train_step
    return model.train_step(*input_args)
  File "/home/ubuntu/venv/potion-voice/voice-cloning/TTS/TTS/tts/models/vits.py", line 1256, in train_step
    outputs = self.forward(
  File "/home/ubuntu/venv/potion-voice/voice-cloning/TTS/TTS/tts/models/vits.py", line 1058, in forward
    pred_embs = self.speaker_manager.encoder.forward(wavs_batch, l2_norm=True)
  File "/home/ubuntu/venv/potion-voice/voice-cloning/TTS/TTS/encoder/models/resnet.py", line 169, in forward
    x = self.torch_spec(x)
  File "/home/ubuntu/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ubuntu/venv/lib/python3.8/site-packages/torch/nn/modules/container.py", line 139, in forward
    input = module(input)
  File "/home/ubuntu/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ubuntu/venv/potion-voice/voice-cloning/TTS/TTS/encoder/models/base_encoder.py", line 22, in forward
    return torch.nn.functional.conv1d(x, self.filter).squeeze(1)
RuntimeError: Input type (**torch.cuda.HalfTensor**) and weight type (**torch.cuda.FloatTensor**) should be the same

I'm training a VITS model over VCTK v0.92.

Cheers,

Markus

**Comment by Ca-ressemble-a-du-fake** [comment]
Hi @MarkuzK,

Sorry I gave up training YourTTS to focus on VITS !

**Comment by MarkuzK** [comment]
Thanks, resolved my issue. Had a mismatch in sampling rate (my pre-trained model was generated from VCTK v.80 with 16k sample_rate) and I was using a higher sample_rate as in the default VITS recipe for VCTK v0.92.

Markus

**Comment by Edresson** [comment]
YourTTS was not easy to reproduce because it needed to change a lot of files and run different scripts. Recently, we created a recipe that makes everything easier. The recipe replicates the first experiment proposed in the YourTTS paper.  The recipe replicates the single language training using the VCTK dataset (it downloads, resamples, and extracts the speaker embeddings automatically :)). However,  if you are interested in multilingual training, we have commented on parameters on the VitsArgs class instance that should be enabled for multilingual training:  https://github.com/coqui-ai/TTS/blob/dev/recipes/vctk/yourtts/train_yourtts.py
